`zig build nvim` fails on Windows (nlua0 exit code 53)

### Problem

When running `zig build nvim` on Windows, the build fails due to the generated `nlua0.exe` exiting with a non-zero exit code.

### Steps to reproduce

```sh
git clone https://github.com/neovim/neovim
cd neovim
zig build nvim
```

### Expected behavior

Build should succeed.

### Nvim version (nvim -v)

0.12.0-dev commit bc0635a9fc9c15a2423d4eb35f627d127d00fa46

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

Windows 11 25H2 (OS Build 26200.7171)

### Terminal name/version

Windows Terminal

### $TERM environment variable

N/A

### Installation

Build from repo

> I am wondering if this is an issue with Zig itself, or an issue with the `build.zig`.

@eamonburns, I guess the issue comes from [libuv/build.zig](https://github.com/allyourcodebase/libuv/blob/dd606c76863c6b3ae38ea2fd7d663e8366e0f90b/build.zig#L62-L63) :)
```zig
            if (optimize == .Debug)
                root_module.linkSystemLibrary("ucrtbased", .{});
```
`optimize` defaults to `Debug`; `ucrtbased.dll` is provided by the Windows SDK and [isn't Redistributable file](https://learn.microsoft.com/en-us/cpp/porting/upgrade-your-code-to-the-universal-crt?view=msvc-170&source=recommendations#where-to-find-the-universal-crt-files). You could try commenting those two lines, or `zig build nlua0 -Doptimize=ReleaseSafe` or
```diff
     if (!system_integration_options.uv) {
-        if (b.lazyDependency("libuv", .{ .target = target, .optimize = optimize })) |dep| {
+        if (b.lazyDependency("libuv", .{ .target = target, .optimize = .ReleaseSafe })) |dep| {
```

As for CMake/MSVC build, I feel the Windows SDK is required too; otherwise, I'd get an error in the first step:
```pwsh
PS C:\neovim\neovim> cmake -S cmake.deps -B .deps -G Ninja -D CMAKE_BUILD_TYPE='RelWithDebInfo'
-- The C compiler identification is MSVC 19.50.35725.0
...
  The C compiler

    "C:/Program Files (x86)/Microsoft Visual Studio/18/BuildTools/VC/Tools/MSVC/14.50.35717/bin/Hostx86/x86/cl.exe"

  is not able to compile a simple test program.
...
    C:\Windows\system32\cmd.exe /C "cd . && "C:\Program Files (x86)\Microsoft Visual Studio\18\BuildTools\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -E vs_link_exe --msvc-ver=1950 --intdir=CMakeFiles\cmTC_e4cd6.dir --rc=rc --mt=CMAKE_MT-NOTFOUND --manifests  -- C:\PROGRA~2\MICROS~2\18\BUILDT~1\VC\Tools\MSVC\1450~1.357\bin\Hostx86\x86\link.exe /nologo CMakeFiles\cmTC_e4cd6.dir\testCCompiler.c.obj  /out:cmTC_e4cd6.exe /implib:cmTC_e4cd6.lib /pdb:cmTC_e4cd6.pdb /version:0.0 /machine:X86  /debug /INCREMENTAL /subsystem:console  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
    RC Pass 1: command "rc /fo CMakeFiles\cmTC_e4cd6.dir/manifest.res CMakeFiles\cmTC_e4cd6.dir/manifest.rc" failed (exit code 0) with the following output:
    no such file or directory
    ninja: build stopped: subcommand failed.
...
```
`rc.exe` and `mt.exe` are also provided by the Windows SDK.

--- a/build.zig
+++ b/build.zig
@@ -102,9 +102,12 @@ pub fn build(b: *std.Build) !void {
         }
     }
     if (!system_integration_options.uv) {
-        if (b.lazyDependency("libuv", .{ .target = target, .optimize = optimize })) |dep| {
+        // NOTE: libuv on Windows depends on Windows SDK when compiled with .Debug mode
+        // https://github.com/neovim/neovim/issues/36889
+        const optimize_uv = if (optimize == .Debug and target.result.os.tag == .windows) .ReleaseSafe else optimize;
+        if (b.lazyDependency("libuv", .{ .target = target, .optimize = optimize_uv })) |dep| {
             libuv = dep.artifact("uv");
-            libluv = try build_lua.build_libluv(b, target, optimize, lua, libuv.?, use_luajit);
+            libluv = try build_lua.build_libluv(b, target, optimize_uv, lua, libuv.?, use_luajit);
 
             libluv_host = if (cross_compiling) libluv_host: {
                 const libuv_dep_host = b.lazyDependency("libuv", .{

--- a/build.zig
+++ b/build.zig
@@ -107,7 +107,7 @@ pub fn build(b: *std.Build) !void {
         const optimize_uv = if (optimize == .Debug and target.result.os.tag == .windows) .ReleaseSafe else optimize;
         if (b.lazyDependency("libuv", .{ .target = target, .optimize = optimize_uv })) |dep| {
             libuv = dep.artifact("uv");
-            libluv = try build_lua.build_libluv(b, target, optimize_uv, lua, libuv.?, use_luajit);
+            libluv = try build_lua.build_libluv(b, target, optimize, lua, libuv.?, use_luajit);
 
             libluv_host = if (cross_compiling) libluv_host: {
                 const libuv_dep_host = b.lazyDependency("libuv", .{

